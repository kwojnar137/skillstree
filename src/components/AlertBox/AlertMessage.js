import React, { useState } from 'react'
import CloseIcon from '@material-ui/icons/Close';
import { AlertBoxContainer } from './AlertBoxContainer'
import { AlertBoxCloseIcon } from './AlertBoxCloseIcon'
import { messegeFromStatus } from '../../utils'


function AlertMessage({ status }) {
  const [show, setShow] = useState(true)

  function handleClose() {
    setShow(false)
  }

  const alertMessage = messegeFromStatus(status)

  if (alertMessage !== 'OK') {
    return (
      <AlertBoxContainer show={show}>
        {alertMessage}
        <AlertBoxCloseIcon >
          <CloseIcon id='closeIcon' onClick={handleClose} />
        </AlertBoxCloseIcon>
      </AlertBoxContainer>
    )
  }
  return <></>
}

export default AlertMessage
