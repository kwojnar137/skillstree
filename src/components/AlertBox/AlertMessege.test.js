import React from 'react';
import { shallow, mount, render } from 'enzyme';
import AlertMessage from './AlertMessage'
import toJson from "enzyme-to-json";


describe("rendering components", () => {
  it('renders without crashing', () => {
    shallow(<AlertMessage />);
  });
})

describe("Rendering status text", () => {
  it('renders alert on status 404', () => {
    const wrapper = shallow(<AlertMessage status={404} />);
    expect(wrapper.contains("Client error: Server couldn't find what you want")).toEqual(true);
  })
  it('renders alert on status 408', () => {
    const wrapper = shallow(<AlertMessage status={408} />);
    expect(wrapper.contains("Client error: Request timeout")).toEqual(true);
  })
  it('renders alert on status 500', () => {
    const wrapper = shallow(<AlertMessage status={500} />);
    expect(wrapper.contains("Internal Server Error. Try again later")).toEqual(true);
  })
  it('renders alert on status 504', () => {
    const wrapper = shallow(<AlertMessage status={504} />);
    expect(wrapper.contains("Unidentified server error. Conntact with our support")).toEqual(true);
  })
})

describe("Click close logic", () => {
  const wrapper = shallow(<AlertMessage status={404} />)
  const closeIcon = wrapper.find("#closeIcon")
  closeIcon.simulate("click")
  // expect(wrapper.find('#closeIcon')).to.toEqual(true)
})

describe("snaphot", () => {
  it('MessegeBox snapshot', () => {
    const boxAlert = shallow(<AlertMessage status={404} />);
    expect(toJson(boxAlert)).toMatchSnapshot();
  })
})


