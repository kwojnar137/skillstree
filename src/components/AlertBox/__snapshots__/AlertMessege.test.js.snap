// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snaphot MessegeBox snapshot 1`] = `
<styled.div
  show={true}
>
  Client error: Server couldn't find what you want
  <styled.div>
    <CloseIcon
      id="closeIcon"
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;
